batchflow/tests/datasetindex_test.py

```diff
@@ -0,0 +1,219 @@
+			dsindex_bad = DatasetIndex([])
+
+	def test_build_index_multidimensional(self):
+		with pytest.raises(TypeError):
+			dsindex_bad = DatasetIndex(np.random.random(size=(SIZE, SIZE)))
+
+
+	#Tests for batchflow.DatasetIndex.get_pos() method
+	def test_get_pos_slice(self, dsindex_list):
+		assert dsindex_list.get_pos(slice(2*SIZE, 3*SIZE, 1)) == slice(0, SIZE, 1)
+		
+	def test_get_pos_str(self):
+		dsindex_str = DatasetIndex(['a', 'b', 'c', 'd', 'e'])
+		assert dsindex_str.get_pos('a') == 0
+		
+	def test_get_pos_str_iterable(self):
+		dsindex_str = DatasetIndex(['a', 'b', 'c', 'd', 'e'])
+		assert set(dsindex_str.get_pos(['a', 'b'])) == set(np.array([0, 1]))
+		assert (dsindex_str.get_pos(['a', 'b']) == np.array([0, 1])).all()
+		
+	def test_get_pos_int(self, dsindex_int):
+		assert dsindex_int.get_pos(SIZE-1) == SIZE-1
+		
+	def test_get_pos_iterable(self, dsindex_list):
+		assert set(dsindex_list.get_pos(range(2*SIZE, 3*SIZE))) == set(range(0, SIZE))
+		assert (dsindex_list.get_pos(range(2*SIZE, 3*SIZE)) == range(0, SIZE)).all() 
+
+
+	#Tests for batchflow.DatasetIndex._shuffle() private method
+	def test_shuffle_bool_false(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=False) 
+		right = np.arange(len(dsindex_list))
+		assert (left == right).all()
+		
+	def test_shuffle_bool_true(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=True)
+		assert (left != np.arange(len(dsindex_list))).any()
+		assert set(left) == set(np.arange(len(dsindex_list)))
+		
+	def test_shuffle_int(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=SIZE)
+		assert (left != np.arange(len(dsindex_list))).any()
+		assert (left == dsindex_list._shuffle(shuffle=SIZE)).all()
+		assert set(left) == set(np.arange(len(dsindex_list)))    
+
+	def test_shuffle_randomstate(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=np.random.RandomState())
+		assert (left != np.arange(len(dsindex_list))).any()
+		left = dsindex_list._shuffle(shuffle=np.random.RandomState(SIZE))
+		right = dsindex_list._shuffle(shuffle=np.random.RandomState(SIZE))
+		assert (left == right).all()
+		left = dsindex_list._shuffle(shuffle=np.random.RandomState())
+		assert set(left) == set(np.arange(len(dsindex_list)))  
+		
+	def test_shuffle_cross(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=SIZE)
+		right = dsindex_list._shuffle(shuffle=np.random.RandomState(SIZE))
+		assert (left == right).all()
+
+	@pytest.mark.skip(reason='permutes index, not order. line 161 of batchflow/base.py')    
+	def test_shuffle_callable(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=np.random.permutation) 
+		assert (left != np.arange(len(dsindex_list))).all()
+		assert set(left) == set(np.arange(len(dsindex_list)))
+
+
+	#Tests for batchflow.DatasetIndex.split() method
+	def test_split_correctness(self, dsindex_big):
+		dsindex_big.split(shares=[0.5, 0.5, 0], shuffle=True)
+		assert len(dsindex_big) == (len(dsindex_big.train) +\
```

Comment:
Fix multiple line statements according to [PEP8](https://pep8.org/#code-lay-out).
Check multilines below too.